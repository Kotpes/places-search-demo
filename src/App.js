//@flow
import React, { Component } from 'react';
import Search from './components/search/Search'
import PlacesList from './components/placesList/PlacesList'
import css from './App.module.css';

const foundLocations = [
  {
    id: "1",
    name: "Woolshed"
  },
  {
    id: "2",
    name: "Aussie Bar"
  },
  {
    id: "3",
    name: "Aussie Bar Club"
  },
  {
    id: "4",
    name: "Baari karpanen"
  }
]

type Location = {
  id: string,
  name: string,
}

type State = {
  filteredLocations: Array<Location>
}
class App extends Component<{}, State> {
  
  state = {
    filteredLocations: foundLocations
  }

  onSearch = (value: string) => {
    const searchName = value.toLowerCase()
    const filteredLocations = foundLocations.filter(item => {
      const name = item.name.toLowerCase()
      return name.includes(searchName)
    })
    this.setState({filteredLocations})    
  }

  render() {
    const {filteredLocations} = this.state
    return (
      <div className={css.app}>
        <h1 className={css.title}>Search for place</h1>
        <Search onChange={(value) => this.onSearch(value)} />
        <PlacesList foundLocations={filteredLocations} />
      </div>
    );
  }
}

export default App;
