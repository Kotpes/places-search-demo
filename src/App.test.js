import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import TestRenderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test(`Should match snapshot`, () => {
  const tree = TestRenderer.create(<App />)
  expect(tree.toJSON()).toMatchSnapshot()
})


