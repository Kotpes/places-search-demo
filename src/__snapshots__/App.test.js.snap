// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div
  role="group"
  style={
    Object {
      "outline": "none",
    }
  }
  tabIndex="-1"
>
  <div
    className="app"
  >
    <h1
      className="title"
    >
      Search for a place
    </h1>
    <section
      className="component"
    >
      <input
        onChange={[Function]}
        placeholder="Start typing"
        type="text"
      />
    </section>
    <div
      className="loading"
    >
      <img
        alt="Loading"
        src="Eclipse-1s-200px.svg"
      />
    </div>
  </div>
</div>
`;
